bootstrap: check commands of openssl exist

It calls openssl commands dhparam and pkcs12 in script bootstrap which
are configurable based on openssl configure options 'no-dh' and
'no-des'. So check existence of these commands. If not, do not run the
script bootstrap.

1. https://github.com/openssl/openssl/blob/master/apps/build.info#L37
2. https://github.com/openssl/openssl/blob/master/apps/build.info#L22

Upstream-Status: Submitted [https://github.com/FreeRADIUS/freeradius-server/pull/4059]

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 raddb/certs/bootstrap | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/raddb/certs/bootstrap b/raddb/certs/bootstrap
index 0f719aafd4..17feddbeeb 100755
--- a/raddb/certs/bootstrap
+++ b/raddb/certs/bootstrap
@@ -13,6 +13,14 @@
 umask 027
 cd `dirname $0`
 
+#  check commands of openssl exis
+for cmd in dhparam pkcs12; do
+	if ! openssl ${cmd} -help >/dev/null 2>&1; then
+		echo "Error: command ${cmd} is not supported by openssl."
+		exit 1
+	fi
+done
+
 make -h > /dev/null 2>&1
 
 #
